src/model/transformations/response-format-simple.js

```diff
@@ -35,10 +65,58 @@ export function remoteToState(remote) {
     [typeName]: datatype,
   };
 }
-
 export function stateToRemote(state, collectedVariables) {
-  const { type: typeName, mandatory, id } = state;
-  const dataType = state[typeName];
+  const { mandatory, id } = state;
+  let { type: typeName } = state;
+  let dataType = state[typeName];
+  
+  if (dataType.format !== undefined && typeName === 'DATE' ) {
+
+    dataType.format = dataType.format.toUpperCase();
+    if(dataType.minimum ===  ''){
+      delete dataType.minimum;
+    }
+
+    if(dataType.maximum ===  ''){
+      delete dataType.maximum;
+    }
+  }
+
+
+    
+  if (typeName === 'DURATION' && dataType.format === 'PnYnM' ){
+    dataType.miyears==''?dataType.miyears = 0: dataType.miyears;
```

Comment:
I will prefer using this syntax for ternary statement
```suggestion
    dataType.miyears = dataType.miyears === '' ? 0 : dataType.miyears;
```